### PR TITLE
Visitor can book property

### LINKED ISSUE #8: Visitor can book a property
As a visitor
When I visit a property show page
And I select the number of guests
And I input a start and end date
And this property is available during those dates
And I click Book Now
Then I should see a message that says "Trip added to cart!"

### PR DESCRIPTION
**Pull request by dtinianow**
closes #8

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -32,7 +32,7 @@ gem 'stripe'
 gem 'paperclip'
 gem 'aws-sdk', '~> 2.3'
 gem 'responders'
-
+gem 'rack_session_access', group: :test
```
Comment (Yoyo2Code) [inline Gemfile]: Great gem add

File: app/views/cart/index.html.erb [hunk 2]
```diff
@@ -1,8 +1,13 @@
...
-    <%= number_to_currency(item.price) %><br>
-    <img src=" <%= item.image_path %> " style="width:220px;height:175px;"/><br><br>
+    <%= cart_item.title %><br>
+    <%= cart_item.description %><br>
+    Subtotal: <%= number_to_currency(cart_item.subtotal) %><br>
+    Total Guests: <%= cart_item.occupancy %><br>
+    Check-In: <%= cart_item.starting_date %><br>
+    Check-Out: <%= cart_item.end_date %><br>
+    <img src="<%= cart_item.image_path %>" style="width:220px;height:175px;"/><br><br>
+    <%#= button_to "Add", cart_items_path({item_id: property.id}) %>
```
Comment (Yoyo2Code) [inline app/views/cart/index.html.erb]: We will need to take out the comments

File: app/models/cart.rb [hunk 3]
```diff
@@ -4,30 +4,40 @@ class Cart
...
 
   def find_items
-    contents.keys.map do |id|
-      Item.find(id)
+    contents.map do |id, data|
+      CartItem.new(id, data["occupancy"], data["starting_date"], data["end_date"])
     end
   end
 
+  # def find_items
```
Comment (Automatic365) [inline app/models/cart.rb]: need to decide if we actually need this. commented out things could pile up pretty quickly.

File: spec/good_tests/features/visitor_can_add_items_to_cart_spec.rb [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    property = location.properties.first
+
+    visit property_path(property, business_name: property.business.slug)
+
+    expect(page).to have_content("Cart Items: 0")
+    # I should see a link or button for "Add to Cart"
+    # When I click "Add to cart" for that item
+    find('#occupancy').find(:xpath, 'option[2]').select_option
+    fill_in :starting_date, with: "08/30/2016"
+    fill_in :end_date, with: "09/05/2016"
```
Comment (dtinianow) [inline spec/good_tests/features/visitor_can_add_items_to_cart_spec.rb]: Still need to create a booked_nights table to account for dates that are already booked
Also need to implement a callback to check for dates already booked

**Comment by Yoyo2Code** [comment]
We can take care of those hound violations later.

**Pull request merged**